### PR TITLE
chore: move virtooltools image builds into the monorepo

### PR DESCRIPTION
**Pull request by igboyes**
## Summary
- Builds `bowtie2`, `cd-hit`, `hmmer`, `pigz`, `samtools`, `seqkit` and `skewer` as their own stages in the root `Dockerfile`, one tool per stage, instead of pulling binaries from the separately-released `ghcr.io/virtool/tools` image — the recipes are that repo's install scripts verbatim.
- Retires the eighth tool, FastQC, outright: `packages/quality-core`, a new Rust crate, reproduces its statistics (quirks included) from a single streaming pass over one FASTQ, pinned bit-for-bit against real 0.11.9 output. This drops the create-sample image's JRE and perl entirely, 886 MB to 491 MB.
- Updates `AGENTS.md` and `docs/images.md` for the one-stage-per-tool rule and the two-crate cargo-chef split, and adds a path-filtered `quality-test` CI job alongside `pathoscope-test`.

### CODE REVIEW

File: packages/quality-core/src/profile.rs [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+/// and it is the honest reading — no A, C, G or T was observed there.
+fn composition_row(cycle: &CycleBases) -> [f64; 4] {
+    let total = cycle.a + cycle.c + cycle.g + cycle.t;
+
+    if total == 0 {
+        return [0.0; 4];
+    }
+
+    [cycle.g, cycle.a, cycle.t, cycle.c]
+        .map(|count| round_half_even((count as f64 / total as f64) * 100.0, 1))
```
[the remark this answers is not in the document]
Comment (igboyes) [reply packages/quality-core/src/profile.rs:341]: Declined — the premise is wrong. FastQC 0.11.9 `PerBaseSequenceContent.getPercentages` divides first:

```java
gPercent[i] = (gCount/(double)total)*100;
aPercent[i] = (aCount/(double)total)*100;
tPercent[i] = (tCount/(double)total)*100;
cPercent[i] = (cCount/(double)total)*100;
```

So for 51 of 80, FastQC itself produces 63.74999999999999, not 63.75, and Python's `round(..., 1)` stores 63.7 — which is what this code already yields. Multiplying by 100 first would store 63.8 and *introduce* the divergence.

Integer-then-divide is `BasicStats`' `%GC`, a different figure, and `gc_percentage` already reproduces that one.

Verified both forms against the frozen FastQC goldens in `tests/fixtures/`: neither reaches a boundary case, so the goldens don't discriminate here — the Java source does.
